wandb/sdk/lib/filesystem.py

```diff
@@ -240,6 +242,97 @@ def safe_copy(source_path: StrPath, target_path: StrPath) -> StrPath:
     return target_path
 
 
+def _reflink_linux(target: StrPath, new_link: StrPath) -> None:
+    """Create a reflink to `target` at `new_link` on Linux."""
+    import fcntl
+
+    FICLONE = 0x40049409  # magic number from <linux/fs.h>  # noqa: N806
+    with open(target, "rb") as t_f, open(new_link, "wb+") as l_f:
+        fcntl.ioctl(l_f.fileno(), FICLONE, t_f.fileno())
+
+
+def _reflink_macos(target: StrPath, new_link: StrPath) -> None:
+    try:
+        clib = ctypes.CDLL("libc.dylib", use_errno=True)
+    except OSError as e:
+        if e.errno != errno.ENOENT:
+            raise
+        clib = ctypes.CDLL("/usr/lib/libSystem.dylib", use_errno=True)
```

Comment:
Why do we try those two paths? Let's add a comment here explaining this.

Reply:
Yeah, only in macOS 10 (Catalina) and earlier is clib a specific file that can be accessed this way.

Reply:
The second path is a fallback for older systems; if we find it at neither path, we can assume it doesn't exist in which case it will raise again, but we won't catch it because there's nothing to do about it.